lib/rb-inotify/notifier.rb

```diff
@@ -22,6 +22,51 @@ module INotify
   #   # Nothing happens until you run the notifier!
   #   notifier.run
   class Notifier
+    # Wrapper around the underlying file handle, to keep track of whether it's
+    # been closed, and ensure it's not leaked if the Notifier goes out of scope
+    # without an explicit #close.
+    #
+    # @private
+    class Handle
```

Comment:
Can you move this into its own file and add documentation to the instance methods?